compiler/src/test/java/com/squareup/anvil/compiler/dagger/ProvidesMethodFactoryGeneratorTest.kt

```diff
@@ -2420,7 +2420,7 @@ public final class DaggerComponentInterface implements ComponentInterface {
       """
     ) {
       val factoryClass = componentInterface.anvilModule
-        .moduleFactoryClass("provideComSquareupTestContributingObjectParentInterface")
+        .moduleFactoryClass("provideParentInterface")
```

Comment:
The new logic produces much easier to read method names.

Reply:
Hmm I was going to leave a comment about avoiding the duplicate name indices by basically using the full package as part of the name and now see that is what we were already doing. Could you clarify the original use-case where you found the issue? Was it two classes/interfaces that also shared the same package?

Reply:
The old method name was `bind${fqName}${boundType.simpleName}`. Now if the bound type uses the same simple name, but different package, then the methods would clash and result in errors. I considered using the FqName for the bound type and it would work, but it looks ugly and is very hard to read. This is way is much better in my opinion.